src/Sentry/Internal/BatchProcessor.cs

```diff
@@ -0,0 +1,99 @@
+using System.Timers;
+using Sentry.Protocol;
+using Sentry.Protocol.Envelopes;
+
+#if NET9_0_OR_GREATER
+using Lock = System.Threading.Lock;
+#else
+using Lock = object;
+#endif
+
+namespace Sentry.Internal;
+
+/// <summary>
+/// The Sentry Batch Processor.
+/// This implementation is not complete yet.
+/// Also, the specification is still work in progress.
+/// </summary>
+/// <remarks>
+/// Sentry Specification: <see href="https://develop.sentry.dev/sdk/telemetry/spans/batch-processor/"/>.
+/// OpenTelemetry spec: <see href="https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md"/>.
+/// </remarks>
+internal sealed class BatchProcessor : IDisposable
+{
+    private readonly IHub _hub;
+    private readonly BatchProcessorTimer _timer;
+    private readonly BatchBuffer<SentryLog> _logs;
+    private readonly Lock _lock;
+
+    private DateTime _lastFlush = DateTime.MinValue;
+
+    public BatchProcessor(IHub hub, int batchCount, TimeSpan batchInterval)
+        : this(hub, batchCount, new TimersBatchProcessorTimer(batchInterval))
+    {
+    }
+
+    public BatchProcessor(IHub hub, int batchCount, BatchProcessorTimer timer)
+    {
+        _hub = hub;
+
+        _timer = timer;
+        _timer.Elapsed += IntervalElapsed;
+
+        _logs = new BatchBuffer<SentryLog>(batchCount);
+        _lock = new Lock();
+    }
+
+    internal void Enqueue(SentryLog log)
+    {
+        lock (_lock)
+        {
+            EnqueueCore(log);
+        }
+    }
+
+    private void EnqueueCore(SentryLog log)
```

Comment:
This method is only called from the internal `Enqueue` method. I guess what it mainly does is avoid a bit of code indentation to make the `Enqueue` method easier to read... in which case it could be made an inline function of that method.

Reply:
Reworked the Enqueue-Approach:
We have two buffers.
If we can't add to the currently active buffer (because it's either full or currently being flushed after a Timeout), we try to add it to the other buffer (after swapping the active buffer, so that upcoming Enqueue-Operations try the new active buffer instead of the previous one being full and/or flushed). Should the second try to enqueue in the other buffer fail as well, we drop the event and add record a discarded event for the Client Report.